src/App.tsx

```diff
@@ -162,55 +165,54 @@ export const App: React.FC = () => {
     todoId: number, completed: boolean,
   ) => {
     await toggleOneTodo(todoId, completed);
-    setChangingTodosId([0]);
     await getTodosFromAPI();
   }, [todos]);
 
   const handleToggleAllTodos = useCallback(async () => {
     try {
       if (counterActiveTodos > 0
-        && fieldForSorting !== FieldForSorting.Completed) {
+        && fieldForFiltering !== FieldForFiltering.Completed) {
         const filteredTodosForToggle = filteredTodos
           .filter(todo => !todo.completed);
 
-        await Promise.all(filteredTodosForToggle.map(async todo => {
-          await toggleOneTodo(todo.id, todo.completed);
-        }));
-      } else {
-        await Promise.all(filteredTodos.map(async todo => {
-          await toggleOneTodo(todo.id, todo.completed);
-        }));
+        await Promise.all(filteredTodosForToggle
+          .map(todo => toggleOneTodo(todo.id, todo.completed)));
+
+        return;
```

Comment:
```suggestion
```
Here you can avoid repeated code. Think about it next way:
1) create constant for your if condition:
```js
const hasNoCompleted = counterActiveTodos > 0
        && fieldForFiltering !== FieldForFiltering.Completed;
```
2) create constant for `filteredTodosForToggle` depends on `hasNoCompleted`:
```js
const filteredTodosForToggle = hasNoCompleted
  ? filteredTodos.filter(todo => !todo.completed)
  : [...filteredTodos];
```
3) use it for Promise.